### DISCUSSION TITLE
What are the oldest python (and, thus linux distribution) versions we should support?

### DISCUSSION THREAD

**Discussion by genodeftest** [exaile/exaile#880] [Polls]
Hi together,

I had a look at some PR validation build and noticed that we are using some linux distributions there that are out of support for some time. When proposing an update to the distributions (exaile/exaile-testimg2#1), @sjohannes mentioned we could directly go to taking Python 3.9 as the oldest supported linux distro. He also wrote:

> For reference, these are the RHEL, Ubuntu (LTS only), and Debian versions that are still supported, with their Python versions:
> |OS|Python version|
> |--|-----|
> |RHEL 7|cbf checking|
> |RHEL 8|3.6|
> |Ubuntu 20.04 (old LTS) |3.8|
> |Debian bullseye (oldstable) |3.9|
> |RHEL 9 (latest) |3.9|
> |Ubuntu 22.04 (latest LTS)|3.10|
> |Debian bookworm (stable)|3.11|

Fedora (both supported versions 37 and 38) and Arch Linux have Python 3.11.

I have rarely contributed to exaile in the last ~2 years so I wouldn't want to take this decision alone, therefore I'm raising this poll.

**Comment by genodeftest** [comment]
I would choose Python 3.9 as anyone using older linux distros doesn't seem to like updating software and thus probably won't update exaile anyways.

**Comment by sjohannes** [comment]
The reason I suggested 3.9 was because it allows us to hit all the latest versions of those "LTS" operating systems. If we ignore RHEL we can jump to 3.10 instead, but I don't think we have a strong reason to do it.

We shouldn't go above 3.10 because that's what Linux Mint uses (it's based on Ubuntu LTS).

**Comment by rokm** [comment]
I'd say we stick with 3.8 as lowest requirement, unless it lacks something that we need. If that's the case, then 3.9 with the same reasoning applied.

**Comment by genodeftest** [comment]
Ok, I'd stay at 3.8 for now. Thanks for taking part in discussion and poll!
